### ISSUE TITLE
False positive ``used-before-assignment``  for walrus operator different lines

### ISSUE THREAD

**Issue by otchita** [PyCQA/pylint#4238]
After the support of the walrus operator, I've noticed that there's still a  `used-before-assignment` false positive when working on different lines:

### Steps to reproduce

Given a file `good-test.py`:
```python
"""Good test example."""
var = f'The number {(count := 4)} is equal to {count}'

```
`pylint` doesn't detect any errors which is expected. However when return to a line like this:

Given a file `bad-test.py`:
```python
"""Bad test example."""
var = f'The number {(count := 4)} ' \
      f'is equal to {count}'

```
We obtain the following:
### Current behavior

Result of `pylint good-test.py`:
```
************* Module bad-test
bad-test.py:2:7: E0601: Using variable 'count' before assignment (used-before-assignment)
```

### Expected behavior

Normally, this should not raise any errors.

### pylint --version output

Result of pylint --version output on my local macOS:

```
pylint 2.7.2
astroid 2.5.1
Python 3.8.8 (v3.8.8:024d8058b0, Feb 19 2021, 08:48:17) 
[Clang 6.0 (clang-600.0.57)]
```

**Comment by jsatt** [comment]
Looks to be similar to #3688, #3763. Also, #4064 seems like an attempt to fix that missed some edge cases.

**Comment by cdce8p** [comment]
This issue seems to be limited to Python 3.8 and below. I've opened #4253 to address it.
